pkg/api/fs.go

```diff
@@ -594,28 +594,43 @@ func (f *Fs) Update(stream pb.Fs_UpdateServer) error {
 		nextVersion = latestVersion + 1
 		logger.Info(ctx, "FS.Update[Init]", key.Project.Field(project), key.Version.Field(nextVersion))
 
+		var deletedPaths []string
+		var objectsToUpdate []*pb.Object
 		for _, object := range objectBuffer {
```

Comment:
Maybe not for this PR but I wonder if we can fold this loop into the one above, we seem to receive all objects and then write them to the DB which made sense so that we didn't block the RPC stream to do DB calls but now that we are splitting and appending to multiple arrays here, maybe we do it as we receive the objects and then do the bulk update/deletes?

Reply:
Yep, no problem at all to do that!